### DISCUSSION TITLE
MQTTX with ThingsSpeak

### DISCUSSION THREAD

**Discussion by Sidlotia** [emqx/MQTTX#900] [Q&A] [answered]
Hi , I am doing this [excample](https://in.mathworks.com/help/thingspeak/use-desktop-mqtt-client-to-publish-to-a-channel.html) , I have configured MQTT X client , it gets connected to my channel .
But when I am doing subscribe  , it get's disconnected , says client not connected?
Also even after entering the required data in topic as per the example it still says topic required, why it is so?

Someone kindly suggest what I am doing wrong in these two steps?

**Comment by ysfscream** [comment]
@Sidlotia Sorry for my late reply.

I carefully read the connection and testing steps in your case, and I don't seem to find any problems. Could you please provide a copy of the available test data for me to debug? Although it looks like it could be an MQTT Broker error and could be checked in the server logs, I mean I would like to reproduce it to see.

**Reply by Sidlotia** [reply]
hi  i am using this public channel for getting my data https://thingspeak.com/channels/12397 .
 
 Still facing the issue ?
 Kindly suggest

**Reply by ysfscream** [reply]
I got this error. How to connect?

<img width="691" alt="image" src="https://user-images.githubusercontent.com/21299158/156088976-dd60a3af-70bc-43ca-8cdf-26ef76316377.png">

**Comment by Sidlotia** [comment, marked as answer]
https://in.mathworks.com/matlabcentral/answers/1659000-i-am-trying-to-publish-data-through-desktop-mqtt-client-but-it-s-get-connected-but-when-i-am-doing 

check this answer, this solves my issue.
Pls in latest update make the topic and data location more clear, it is very deceptive

**Reply by ysfscream** [reply]
Glad you solved this issue.

But I don't think it's a matter of not distinguishing between the topic and payload inputs, perhaps a brief reading of the documentation will help you: https://mqttx.app/docs#quickly-establish-a-connection
